### PR TITLE
chore(deps): bumps alloy, revm, fork-db

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Alloy `0.5.2` is out.

## Solution

Bump alloy + revm and related deps such as fork-db and op-alloy.

### CODE REVIEW

**Comment by DaniPopes** [comment]
Can we add a test for https://github.com/foundry-rs/foundry/issues/9046 and close the issue in this PR?

**Comment by yash-atreya** [comment]
> Can we add a test for #9046 and close the issue in this PR?

alloy `0.5.2` doesn't include a fix for #9046. See https://github.com/alloy-rs/alloy/pull/1496

File: crates/anvil/src/hardfork.rs [hunk 1]
```diff
@@ -135,7 +135,7 @@ impl From<EthereumHardfork> for SpecId {
             EthereumHardfork::Cancun | EthereumHardfork::Latest => Self::CANCUN,
             EthereumHardfork::Prague => Self::PRAGUE,
             // TODO: switch to latest after activation
-            EthereumHardfork::PragueEOF => Self::PRAGUE_EOF,
+            EthereumHardfork::PragueEOF => Self::PRAGUE,
```
Comment (zerosnacks) [inline crates/anvil/src/hardfork.rs:138]: Could you expand on the origin of this change? I'm missing some context

It looks like EOF is not going to part of the first Prague hardfork

File: crates/config/src/utils.rs [hunk 2]
```diff
@@ -301,7 +301,7 @@ impl FromStr for Numeric {
 #[inline]
 pub fn evm_spec_id(evm_version: &EvmVersion, alphanet: bool) -> SpecId {
     if alphanet {
-        return SpecId::PRAGUE_EOF;
```
Comment (zerosnacks) [inline crates/config/src/utils.rs:304]: Same here, why is this changed? Odyssey supports EOF

File: crates/anvil/src/hardfork.rs (hunk 1 above)
Comment (yash-atreya) [reply crates/anvil/src/hardfork.rs:138]: Fixing PragueEOF was renamed to OSAKA

**Review by mattsse — APPROVED** [review]
